BUG: `ScopingPolicy` members improperly typed

Description
----

Somehow `mypy` isn't recognizing that the `ScopingPolicy` members are its own instances and are hence strings; maybe it has to do with the hacky backport we have for `enum.StrEnum` (`line_profiler.line_profiler_utils._StrEnumBase`). 

### Example script (`scoping-policy-type-bug.py`)

```python
from line_profiler.line_profiler import LineProfiler
from line_profiler.scoping_policy import ScopingPolicy


class MyClass:
    pass


def main() -> None:
    prof = LineProfiler()
    prof.add_class(MyClass, scoping_policy=ScopingPolicy.NONE)
```

### Output

```shell
$ mypy scoping-policy-enum-bug.py
scoping-policy-type-bug.py:11 error: argument "scoping_policy" to "add_class" of "LineProfiler" has incompatible type "auto"; expected "ScopingPolicy | str | ScopingPolicyDict | None"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

Mitigation
----

I noticed that in #419 we're circumventing this within the codebase with manual type-casting (`cast(ScopingPolicy, ScopingPolicy.<attr>)`; see d1b4c23). In some sense this is alright, since end users are more likely to just use the string forms for `LineProfiler.add_*(..., scoping_policy=<policy>)`... but we should probably fix this more thoroughly since [`ScopingPolicy` is public API](https://kernprof.readthedocs.io/en/latest/auto/line_profiler.scoping_policy.html#line_profiler.scoping_policy.ScopingPolicy), and having the imported enum class not work would be quite counterintuitive.

Wonder if something like
```python
class _StrEnumBase(str, enum.Enum):
    ...


if TYPE_CHECKING:
    from enum import StrEnum as _StrEnumBase  # type: ignore[attr-defined]


class StringEnum(_StrEnumBase):
    ...
```
in `line_profiler/line_profiler_utils.py` would work... if not, we may have to use the [`enum.auto`-subclassing workaround](https://stackoverflow.com/questions/69318005/mypy-gives-an-incompatible-type-auto-expected-str-error-when-using-auto).

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -7,6 +7,9 @@ Changes
   ``sys.modules['__main__']`` namespace to avoid issues w/e.g. pickling
   (#423)
 * CHANGE: Drop support for Python 3.8 and Python 3.9
+* FIX: ``ScopingPolicy`` members now type-check properly as instances
+  thereof -- at least on Python versions (3.11+) where ``enum.StrEnum``
+  is available (#427)
 
 
 5.0.2

--- a/line_profiler/line_profiler.py
+++ b/line_profiler/line_profiler.py
@@ -600,15 +600,9 @@ class LineProfiler(CLineProfiler, ByCountProfilerMixin):
         namespace: type | types.ModuleType,
         *,
         seen: set[int] | None = None,
-        func_scoping_policy: ScopingPolicy = cast(
-            ScopingPolicy, ScopingPolicy.NONE
-        ),
-        class_scoping_policy: ScopingPolicy = cast(
-            ScopingPolicy, ScopingPolicy.NONE
-        ),
-        module_scoping_policy: ScopingPolicy = cast(
-            ScopingPolicy, ScopingPolicy.NONE
-        ),
+        func_scoping_policy: ScopingPolicy = ScopingPolicy.NONE,
+        class_scoping_policy: ScopingPolicy = ScopingPolicy.NONE,
+        module_scoping_policy: ScopingPolicy = ScopingPolicy.NONE,
         wrap: bool = False,
         name: str | None = None,
     ) -> int:

--- a/line_profiler/line_profiler_utils.py
+++ b/line_profiler/line_profiler_utils.py
@@ -46,7 +46,14 @@ class _StrEnumBase(str, enum.Enum):
         return self.value
 
 
-class StringEnum(getattr(enum, 'StrEnum', _StrEnumBase)):  # type: ignore[misc]
+try:
+    from enum import StrEnum as _StrEnum
+except ImportError:
+    if not typing.TYPE_CHECKING:  # Don't confuse the typechecker
+        _StrEnum = _StrEnumBase
+
+
+class StringEnum(_StrEnum):
     """
     Convenience wrapper around :py:class:`enum.StrEnum`.
 

